### PR TITLE
Support other networks for get-sns script

### PR DESCRIPTION
**Pull request by dskloetd**
# Motivation

`scripts/sns/aggregator/get-sns` is a script to conveniently get some information about any SNS based on the name or any of its canister IDs.

Currently it only works with mainnet SNSes but while trying to reproduce out-of-cycles SNSes locally, I found it useful to be able to use this locally.

# Changes

1. Add `--canister` flag to `scripts/dfx-server-origin` to support adding a canister as the subdomain in the origin. This is useful to create an SNS aggregator URL.
2. Add `--network` flag to `scripts/sns/aggregator/get-sns`.
3. Download aggregator pages if the network is not `mainnet` (if it is `mainnet`, then we use the pages present in the repo).
4. Connect to the aggregator based on the `--network` flag to get the current data.

# Tests

1. I used the script manually to get data from `mainnet`, `local` and `devenv_dskloet`.
2. Install `dfx` on CI because it's used by `scripts/dfx-server-origin`.
3. The [CI test](https://github.com/dfinity/nns-dapp/blob/c6e4e8f3aa470282d5e6a32b30a79758385457cc/.github/workflows/checks.yml#L318) still passes.

# Todos

- [ ] Add entry to changelog (if necessary).
not necessary

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -287,6 +287,8 @@ jobs:
             # moreutils is needed for sponge, used by network-config.test
             brew install bash coreutils moreutils
           fi
+      - name: Install dfx
```
Comment (mstrasinskis) [inline .github/workflows/checks.yml:290]: Is this related to the pr?

Comment (dskloetd) [reply .github/workflows/checks.yml:290]: Yes, `scripts/dfx-server-origin` uses `dfx` so it now needs to be installed here.
Thanks, I forgot to add that to the PR description. I have now added it.

Comment (mstrasinskis) [reply .github/workflows/checks.yml:290]: Thanks for the explanation.

**Review by mstrasinskis — APPROVED** [review]
Thanks!
